### PR TITLE
assignment2

### PR DESCRIPTION
**Pull request by meta-shubham-joshi**

### CODE REVIEW

File: PF/PF-Assignment2/FixedProductPromotion.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import java.util.Calendar;
+
+interface Promotion{
+	enum PromotionEnum{
+		get5("GET05",1,5),free5("FREE05",15,20),less5("LESS05",25,30);  // @params promotion code,start and end date of in a month 
+		String promoCode;
+		double discount;
+		int startDate,endDate;
```
Comment (Hitesh-Khatri-Metacube) [inline PF/PF-Assignment2/FixedProductPromotion.java]: Use Date type when need to compare dates.

File: PF/PF-Assignment2/FixedProductPromotion.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	public void setFixedDiscount(){
+		fixedDiscount = 0.05;	// 5% discount is given 
+	}
+	
+	public boolean isPromotionApplicable(int productInBill,double billAmount, String userCode){ // @params no of products in bill, total amount, promo code entered
+		Calendar cal = Calendar.getInstance();													// @return boolean variable to show discount availability
+		int dayOfMonth = cal.get(Calendar.DAY_OF_MONTH);
+		if(billAmount>=getMinimumPrice())
+			if(productInBill>=getMinQuantity()){
+				for(PromotionEnum promo : PromotionEnum.values())
```
Comment (Hitesh-Khatri-Metacube) [inline PF/PF-Assignment2/FixedProductPromotion.java]: Use curly braces with loop, control, conditional statement when logic is taking more than one lines. to improve the code readability.

File: PF/PF-Assignment2/FixedOrderPromotion.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	public void setMinimumPrice(double price){		// setting minimum price
+		minimumPrice = price;
+	}
+	
+	public double getFixedDiscount(){				// retrieving fixed discount
+		return fixedDiscount;
+	}
+	
+	public void setFixedDiscount(){
+		fixedDiscount = 0.05;						// 5% discount is given
```
[the remark this answers is not in the document]
Comment (meta-shubham-joshi) [reply PF/PF-Assignment2/FixedOrderPromotion.java]: changed the setFixedDiscount method by taking the value in the parameter and then assigning it to the class variable.

File: PF/PF-Assignment2/FixedProductPromotion.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import java.util.Calendar;
+
+interface Promotion{
```
[the remark this answers is not in the document]
Comment (meta-shubham-joshi) [reply PF/PF-Assignment2/FixedProductPromotion.java]: Added access modifiers at appropriate positions.

File: PF/PF-Assignment2/FixedOrderPromotion.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import java.util.Calendar;
+
+public class FixedOrderPromotion implements Promotion{		// class implementing interface Promotion
```
[the remark this answers is not in the document]
Comment (meta-shubham-joshi) [reply PF/PF-Assignment2/FixedOrderPromotion.java]: Added comments appropriately.

File: PF/PF-Assignment2/FixedProductPromotion.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	public void setMinimumPrice(int quantity, double price){
+		minQuantity = quantity;
+		minimumPrice = price;
+	}
+	
+	public double getFixedDiscount(){
+		return fixedDiscount;
+	}
+	
+	public void setFixedDiscount(){
```
[the remark this answers is not in the document]
Comment (meta-shubham-joshi) [reply PF/PF-Assignment2/FixedProductPromotion.java]: Changed accordingly.

**Pull request merged**